crates/c-api/include/wasmtime.h

```diff
@@ -38,6 +38,12 @@ WASMTIME_CONFIG_PROP(cranelift_opt_level, wasmtime_opt_level_t)
 
 ///////////////////////////////////////////////////////////////////////////////
 
+bool wasmtime_wat2wasm(
+    wasm_engine_t *engine,
+    const wasm_byte_vec_t *wat,
+    own wasm_byte_vec_t *ret,
```

Comment:
Since this is our own API, perhaps this should also return an (output param) error message when it returns false? Another `wasm_byte_vec_t` as a UTF-8 string, perhaps?

Reply:
Seems reasonable to me!